src/lifetracker/logic/LogicImpl.java

```diff
@@ -49,18 +40,30 @@ public ExecuteResult executeCommand(String commandString) {
 
         if (commandString.equals("exit")) {
             return runResult;
+
         } else {
             CommandObject commandToExecute;
             CalendarList executedState;
 
-            try {
+            if (commandString.equals("undo")) {
+
+                commandStack.pop();
```

Comment:
Imagine what happens here. The user inputs "add meeting", then "undo". The code here will just execute "add meeting" again, which is wrong.

What should happen is that you store the CommandObject into the stack. When "undo" is called, pop out a CommandObject, then call it's undo function.